core/include/tee/tadb.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2017, Linaro Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#ifndef __TEE_TADB_H
+#define __TEE_TADB_H
+
+#include <tee/tee_fs.h>
+
+struct tee_tadb_ta;
+
+/*
+ * struct tee_tadb_property
+ * @uuid:	UUID of Trusted Application (TA) or Security Domain (SD)
+ * @parent:	UUID of parent SD, zeroes if nonexistent
+ * @handler:	UUID of TA handling this SD
+ * @version:	Version of TA or SD
+ * @custom_size:Size of customized properties, prepended to the encrypted
+ *		TA binary
+ * @bin_size:	Size of the binary TA
+ * @is_ta:	True if this is a TA
+ * @is_root:	True if this is a root SD or a TA without an SD
+ */
+struct tee_tadb_property {
+	TEE_UUID uuid;
+	TEE_UUID parent;
+	TEE_UUID handler;
+	uint32_t version;
+	uint32_t custom_size;
+	uint32_t bin_size;
+	bool is_ta;
+	bool is_root;
```

Comment:
Since `parent`, `handler`, `is_ta` and `is_root` are currently not used, I'd rather introduce them later. It is fine to mention in the commit log that this may be a first step towards supporting SDs, but it should not show in the code yet. The "secure storage TA" feature is useful on its own.

Reply:
OK